### PR TITLE
fix #17726 Databricks schema name with hyphen issue

### LINKED ISSUE #17726: Databricks connector fails to ingest tables under the schema with dashes and undescore
**Affected module**
Ingestion Framework

**Describe the bug**
Ingestions failed to get tables under the schema that have both - and _ in the name (e.g.: test_schema-name). Schema names like these need to be wrapped with back-quotes as `test_schema-name` in the query.

**To Reproduce**
The Table can not be injested under schema name test_schema-name, Because the schema name contains both - and _.

image

Here are the logs
```
== SQL ==
/* {"app": "OpenMetadata", "version": "1.5.2.0"} */
DESCRIBE TABLE EXTENDED test_schema-name.test_table_1
---------------------------^^^
[SQL: /* {"app": "OpenMetadata", "version": "1.5.2.0"} */
DESCRIBE TABLE EXTENDED test_schema-name.test_table_1]
(Background on this error at: https://sqlalche.me/e/14/4xp6)
[2024-09-05, 07:02:05 UTC] {metadata.py:330} ERROR - Failed to fetch table type for table test_table_1 due to: (databricks.sql.exc.ServerOperationError) 
[INVALID_IDENTIFIER] The identifier test_schema-name is invalid. Please, consider quoting it with back-quotes as `test_schema-name`. SQLSTATE: 42602 (line 2, pos 27)
== SQL ==

```

**Expected behavior**
This ingestion should work, it's a valid schema name.

**Version:**
 - OpenMetadata version: 1.5.2 
 - OpenMetadata Ingestion package version: 1.5.2      
 
image

**Additional context**
For more info you can find in Slack https://openmetadata.slack.com/archives/C02B6955S4S/p1725442179876779

### PR DESCRIPTION
**Pull request by akashverma0786**
### Describe your changes:

Fixes #17726 

Below Image shows schema name "test_schema-name" has tables ingested
Screenshot 2024-11-13 at 8 39 03 PM

Below image shows the logs
Screenshot 2024-11-13 at 8 38 46 PM

#
### Type of change:

- [x] Bug fix
- [ ] Improvement
- [ ] New feature
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Documentation

#
### Checklist:

- [x] I have read the [**CONTRIBUTING**](https://docs.open-metadata.org/developers/contribute) document.
- [x] My PR title is `Fixes <issue-number>: <short explanation>`
- [ ] I have commented on my code, particularly in hard-to-understand areas. 
- [ ] For JSON Schema changes: I updated the migration scripts or explained why it is not needed.

### CODE REVIEW

File: ingestion/src/metadata/ingestion/source/database/databricks/metadata.py [hunk 1]
```diff
@@ -111,9 +111,11 @@ class MAP(String):
         "binary": create_sqlalchemy_type("BINARY"),
     }
 )
+def validate_schema(schema):
+    # Adds back quotes(``) if hyphen(-) in schema name 
+    return f"`{schema}`" if "-" in schema else schema
+
 
-# This method is from hive dialect originally but
-# is overridden to optimize DESCRIBE query execution
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/database/databricks/metadata.py:116]: let's not remove the comments

**Comment by ayush-shah** [comment]
Let's add test around here

File: ingestion/src/metadata/ingestion/source/database/databricks/metadata.py [hunk 2]
```diff
@@ -112,8 +112,12 @@ class MAP(String):
     }
 )
 
-# This method is from hive dialect originally but
-# is overridden to optimize DESCRIBE query execution
```
Comment (ayush-shah) [inline ingestion/src/metadata/ingestion/source/database/databricks/metadata.py:116]: this has not been resolved, let's revert the comment deletion

Comment (akashverma0786) [reply ingestion/src/metadata/ingestion/source/database/databricks/metadata.py]: looks good right?

**Review by ayush-shah — APPROVED** [review]

**Pull request merged**